### DISCUSSION TITLE
Force add alpha channel when converting from png to bmp3?

### DISCUSSION THREAD

**Discussion by Agendum** [ImageMagick/ImageMagick#3683] [Help]
I am using the ImageMagick NuGet library in C# to convert a png into a bmp3. The target file must be 32-bit (four channels, 8 bits-per-channel), and the alpha can just be solid white. I have tried this:

```C#
var fromImage = new MagickImage(fromFilePath);
fromImage.Format = MagickFormat.Bmp3;
fromImage.Settings.Format = MagickFormat.Bmp3;
fromImage.Alpha(AlphaOption.On);
fromImage.HasAlpha = true;
fromImage.ColorAlpha(new MagickColor(0xff, 0xff, 0xff));
fromImage.ColorType = ColorType.TrueColorAlpha;
fromImage.Settings.ColorType = ColorType.TrueColorAlpha;
fromImage.Write(toFilePath);
```

But the file is always 24-bit. The source png has these identity details:

```
  Colorspace: sRGB
  Type: TrueColorAlpha
  Depth: 8-bit
  Channel depth:
    Red: 8-bit
    Green: 8-bit
    Blue: 8-bit
    Alpha: 1-bit
```

And the target file always ends up being:

```
  Colorspace: sRGB
  Type: TrueColor
  Depth: 8-bit
  Channel depth:
    Red: 8-bit
    Green: 8-bit
    Blue: 8-bit
```

Is there something I can do to force ImageMagick to create an 8-bit alpha channel, with white color?

Thanks!

**Comment by snibgo** [comment]
BMP3 can't record alpha. Use just "BMP".

**Reply by Agendum** [reply]
Ah thanks, that is what I was missing.
